File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extract() {
    try {
        let dataBuffer = fs.readFileSync('School LMS.pdf');
        let data = await pdf(dataBuffer);
        fs.writeFileSync('.gemini_pdf_extract.txt', data.text);
        console.log('PDF extracted to .gemini_pdf_extract.txt');
    } catch(err) {
        console.error(err);
    }
}
extract();

File: extract_pdf.py
import sys

def extract():
    try:
        import PyPDF2
        reader = PyPDF2.PdfReader('School LMS.pdf')
        text = ""
        for page in reader.pages:
            text += page.extract_text() + "\n"
        with open('.gemini_pdf_extract.txt', 'w', encoding='utf-8') as f:
            f.write(text)
        print("Extracted successfully")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    extract()
